nmdc_schema/migrators/migrator_from_11_8_0_to_11_9_0.py

```diff
@@ -0,0 +1,100 @@
+            ms_id = workflow_execution_record.get("id")
+            dg_id = workflow_execution_record.get("was_informed_by")
+            dg_record = {
+                "ms_id": ms_id,
+                "started_at_time": workflow_execution_record.get("started_at_time"),
+                "ended_at_time": workflow_execution_record.get("ended_at_time"),
+                "has_output": workflow_execution_record.get("has_output", []),
+            }
+            self.data_objects_mapping[dg_record["has_output"][0]] = workflow_execution_record.get("was_informed_by")
+            # Add the dict to the wfe_mappings
+            self.wfe_mappings[dg_id] = dg_record
+
+    def link_do_to_dg(self, data_object_record: dict) -> dict:
+        r"""
+        For each data object, link it to the corresponding DataGeneration record through the was_generated_by field using the data_objects_mapping.
+        """
+
+        data_object_ids = self.data_objects_mapping.values()
+        dg_id = data_object_record.get("id")
+        # Check if the data object is an object of interest
+        if dg_id in data_object_ids:
+            # Check if the data object already has a was_generated_by field
+            if data_object_record.get("was_generated_by") == None:
+                data_object_record["was_generated_by"] = self.data_objects_mapping.get(dg_id)
+            else:
+                # If it already has a was_generated_by field, we do not want to overwrite it
+                self.logger.warning(f"Data object {dg_id} already has a was_generated_by field, skipping update.")
+        return data_object_record
+
+    def migrate_fields_to_dg(self, data_generation_record: dict) -> dict:
+        r"""
+        Migrate the fields from the workflow execution record to the DataGeneration record.
+        """
+        # Get the target ids from the wfe_mappings
+        target_ids = list(self.wfe_mappings.keys())
+        dg_record_id = data_generation_record.get("id")
+
+        # Check if the data generation record is one of the target ids
+        if dg_record_id in target_ids:
+            # get the corresponding mapping from wfe_mappings
+            wfe_map = self.wfe_mappings[dg_record_id]
+            dg_has_output = data_generation_record.get("has_output",[])
+            wfe_map_has_output = wfe_map.get("has_output", [])
+            # Migrate the time fields
+            data_generation_record["start_time"] = wfe_map.get("started_at_time", None)
+            data_generation_record["end_time"] = wfe_map.get("ended_at_time", None)
+            
+            # Migrate the has_output field
+            if not dg_has_output:
+                data_generation_record["has_output"] = wfe_map_has_output
+
+            elif dg_has_output and not any(item in dg_has_output for item in wfe_map_has_output):
+                data_generation_record["has_output"] = data_generation_record["has_output"] + wfe_map_has_output
+                
```

Comment:
Might be worth rewriting this part to combine the two lists (wfe_map_has_output and dg_has_output), then list(set()) the combined lists?  That would mean fewer conditionals and might fix your has_output issue.